src/consensus/encode.rs

```diff
@@ -240,7 +240,6 @@ macro_rules! decoder_fn {
     ($name:ident, $val_type:ty, $readfn:ident, $byte_len: expr) => {
         #[inline]
         fn $name(&mut self) -> Result<$val_type, Error> {
-            debug_assert_eq!(::core::mem::size_of::<$val_type>(), $byte_len); // size_of isn't a constfn in 1.22
```

Comment:
Why remove this?
Also, I'm pretty sure after the MSRV bump we can make this a compile-time check:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=44cce9ee0d3113cdaaaf7c6f421d5699

Reply:
Cool. I'm no macro expert but is this as good or am I missing something?
```
macro_rules! const_assert {
    ($x:expr) => {
        const _UNUSED: [(); 0 - !$x as usize] = [];
    };
}
```
This works with Rust 1.29 as well.

Reply:
OoO I didn't think that this can be done in 1.29! 
That will work but will limit the macro to work only once in every scope, otherwise, you'll get multiple definitions of `_UNUSED`. but it doesn't seem like we ever const-assert more than one thing in the same scope

Reply:
Just to double check, is the simplification of the original 
```rust
[(); 0 - !{ const ASSERT: bool = $x; ASSERT } as usize] = []
``` 
to
```rust
[(); 0 - !$x as usize] = []
```
ok? I'm going to force push this change but wanted to explicitly double check with you. Thanks

Reply:
Yes :)